### PR TITLE
chore: enable github bot CI rerun feature

### PR DESCRIPTION
**Pull request by josephperrott**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

**Review by ocombe — APPROVED** [review]

**Comment by brandonroberts** [comment]
Will there be a trigger to only re-run failed jobs?

File: .github/angular-robot.yml [hunk 1]
```diff
@@ -166,3 +166,10 @@ triagePR:
       - "effort*"
       - "risk*"
       - "comp: *"
+
+# options for rerunning CI
+rerunCircleCI:
+  # set to true to disable
+  disabled: false
+  # the label which when added triggers a rerun of the default CircleCI workflow
+  triggerRerunLabel: 'Rerun CI at HEAD'
```
Comment (gkalpak) [inline .github/angular-robot.yml]: What does the `at HEAD` mean? How is it different than `Rerun CI`?

**Comment by josephperrott** [comment]
@brandonroberts the plan is to add this in the future, but the functionality for it will need to be added to the github robot first.

File: .github/angular-robot.yml (hunk 1 above)
Comment (josephperrott) [reply .github/angular-robot.yml]: @IgorMinar wanted the label to reflect that the CI will run at the current HEAD of the targeted branch being merged into, usually master.

**Review by IgorMinar — APPROVED** [review]

File: .github/angular-robot.yml (hunk 1 above)
Comment (filipesilva) [reply .github/angular-robot.yml]: All our CI runs at the HEAD of its target branch though:

https://github.com/angular/angular/blob/5b134caf2dc57ccb9d6c34ef39227f922b0aa36d/.circleci/config.yml#L66-L79

File: .github/angular-robot.yml [hunk 2]
```diff
@@ -167,3 +167,10 @@ triagePR:
       - "effort*"
       - "risk*"
       - "comp: *"
+
+# options for rerunning CI
+rerunCircleCI:
+  # set to true to disable
+  disabled: false
+  # the label which when added triggers a rerun of the default CircleCI workflow
+  triggerRerunLabel: "PR action: rerun CI at HEAD"
```
Comment (gkalpak) [inline .github/angular-robot.yml]: 🤔 `PR action:` labels usually indicate that someone needs to do something. `PR action: rerun CI at HEAD` sounds like someone needs to do it (vs it is done automatically for us).

Also, this label is not documented in [docs/TRIAGE_AND_LABELS.md][1] 😞 

[1]: https://github.com/angular/angular/blob/537502d6855bb184571b95d8f1ceb1484071ff94/docs/TRIAGE_AND_LABELS.md

Comment (gkalpak) [reply .github/angular-robot.yml]: For reference: Documented it in #30171.

**Pull request closed without merging**